### PR TITLE
MOB 8826 - limit the number of stored events and make the number of s…

### PR DESCRIPTION
**Pull request by megha-iterable**
## Jira Ticket(s)

* [MOB-8826 ](https://iterable.atlassian.net/browse/MOB-8826)

## Description

>  limit the number of stored events and make the number of synced events configurable.

## Test Steps

> Video attached to the ticket, See a JIRA ticket for the video

### CODE REVIEW

File: swift-sdk/Internal/AnonymousUserManager.swift [hunk 1]
```diff
@@ -204,6 +205,11 @@ public class AnonymousUserManager: AnonymousUserManagerProtocol {
         } else {
             eventsDataObjects.append(appendData)
         }
+
+        let eventDataCount = eventsDataObjects.count
+        if  eventDataCount > config.eventThresholdLimit {
+            eventsDataObjects = eventsDataObjects.suffix(config.eventThresholdLimit)
```
Comment (evantk91) [inline swift-sdk/Internal/AnonymousUserManager.swift:211]: Right now the number of events stored is one more than the threshold passed in on the config.

File: swift-sdk/IterableConfig.swift [hunk 2]
```diff
@@ -132,4 +132,7 @@ public class IterableConfig: NSObject {
     public var enableAnonTracking = true
     /// Allows for fetching embedded messages.
     public var enableEmbeddedMessaging = false
+
+    // How many events can be stored in the local storage. By default limt is 100.
+    public var eventThresholdLimit: Int = 100
```
Comment (evantk91) [inline swift-sdk/IterableConfig.swift:137]: The default threshold should be updated to 50.

**Review by evantk91 — COMMENTED** [review]
Left some comments.

File: swift-sdk/IterableConfig.swift (hunk 2 above)
Comment (davidtruong) [reply swift-sdk/IterableConfig.swift:137]: Updated requirements to 100.

File: swift-sdk/Internal/AnonymousUserManager.swift (hunk 1 above)
Comment (evantk91) [reply swift-sdk/Internal/AnonymousUserManager.swift:211]: I will update this and merge.

Comment (evantk91) [reply swift-sdk/Internal/AnonymousUserManager.swift:211]: Nevermind this works.

**Pull request merged**